frontend/src/scenes/surveys/surveyActivityDescriber.tsx

```diff
@@ -149,161 +211,125 @@ const surveyActionsMapping: Record<
         const afterConditions = change?.after as Survey['conditions']
         const changes: JSX.Element[] = []
 
-        if (!isEmptyOrUndefined(beforeConditions?.url) || !isEmptyOrUndefined(afterConditions?.url)) {
-            if (isEmptyOrUndefined(beforeConditions?.url) && !isEmptyOrUndefined(afterConditions?.url)) {
-                changes.push(
-                    <>
-                        set URL condition to <strong>{afterConditions?.url}</strong>
-                    </>
-                )
-            } else if (!isEmptyOrUndefined(beforeConditions?.url) && isEmptyOrUndefined(afterConditions?.url)) {
-                changes.push(
-                    <>
-                        removed URL condition (was <strong>{beforeConditions?.url}</strong>)
-                    </>
-                )
-            } else if (beforeConditions?.url !== afterConditions?.url) {
-                changes.push(
-                    <>
-                        changed URL condition from{' '}
-                        {!isEmptyOrUndefined(beforeConditions?.url) ? (
-                            <strong>{beforeConditions?.url}</strong>
-                        ) : (
-                            <i>unset</i>
-                        )}{' '}
-                        to <strong>{afterConditions?.url}</strong>
-                    </>
-                )
-            }
+        changes.push(
+            describeFieldChange('URL condition', beforeConditions?.url, afterConditions?.url),
+            describeFieldChange('selector', beforeConditions?.selector, afterConditions?.selector),
+            describeFieldChange(
+                'wait period',
+                beforeConditions?.seenSurveyWaitPeriodInDays,
+                afterConditions?.seenSurveyWaitPeriodInDays,
+                'days'
+            ),
+            describeFieldChange('URL match type', beforeConditions?.urlMatchType, afterConditions?.urlMatchType)
+        )
+
+        // Use JSON.stringify for deep comparison of objects
+        if (JSON.stringify(beforeConditions?.events) !== JSON.stringify(afterConditions?.events)) {
+            changes.push(<>modified event conditions</>)
```

Comment:
elected to not go into details about the event condition changes, just letting users know that they were changed at all.

Reply:
reasonable for now 👍